tests/test_optimize.py

```diff
@@ -462,6 +468,125 @@ def test_adam_optimizer_multivar(self, bunch, tol):
                 x_twosteps_target = x_onestep - adapted_stepsize * firstmoment / (np.sqrt(secondmoment) + 1e-8)
                 assert x_twosteps == pytest.approx(x_twosteps_target, abs=tol)
 
+    @pytest.mark.parametrize('x_start', x_vals)
+    def test_rotosolve_optimizer_univar(self, x_start, bunch, tol):
+        """Tests that rotosolve optimizer takes one and two steps correctly
+        for uni-variate functions."""
+
+        for f in univariate_funcs:
+            x_onestep = bunch.rotosolve_opt.step(f, x_start)
+            x_onestep_target = self.rotosolve_step(f, x_start)
```

Comment:
A comment:

I know that the other optimizer tests are working similarly, i.e. by comparing the result of the actual optimizer to a routine in the test that reproduces the recipe of the optimization. But in the meantime we realised that tests should use cases for which we can easily calculate actual numbers, and compare with these numbers as targets.

I think this is quite a bit of work, but in the long run we should think of refactoring the optimizer tests this way...

Reply:
I agree that it seems better to compare to known solutions. In this case, as you noted, I simply implemented the new tests in the same way as the other ones. It might be better to change all tests to test the optimizers against analytically calculated answers in a new PR?